src/Framework/ResultCacheListener.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework;
+
+use PHPUnit\Runner\TestResultCache;
+use PHPUnit\Util\Printer;
+
+class ResultCacheListener extends Printer implements TestListener
+{
+    public const SUITE_START     = 'suite_start';
+    public const SUITE_END       = 'suite';
+    public const TEST_START      = 'start';
+    public const TEST_PASSED     = 'passed';
+    public const TEST_SKIPPED    = 'skipped';
+    public const TEST_ERROR      = 'error';
+    public const TEST_WARNING    = 'warning';
+    public const TEST_FAILURE    = 'failure';
+    public const TEST_RISKY      = 'risky';
+    public const TEST_INCOMPLETE = 'incomplete';
+    public const TEST_UNKNOWN    = 'unknown';
+
+    private $runState = [];
+
+    /**
+     * @var TestResultCache
```

Comment:
```
new ResultCacheListener()
```

boom ! this property is `null`, not `TestResultCache` ;)

Reply:
added some phpdoc ...which I then even had to sort 🤓